pyomo/solvers/plugins/solvers/GAMS.py

```diff
@@ -0,0 +1,792 @@
+#  Copyright 2017 National Technology and Engineering Solutions of Sandia, LLC
+#  Under the terms of Contract DE-NA0003525 with National Technology and 
+#  Engineering Solutions of Sandia, LLC, the U.S. Government retains certain 
+#  rights in this software.
+#  This software is distributed under the 3-clause BSD License.
+#  ___________________________________________________________________________
+
+from __future__ import print_function
+from six import StringIO
+from pyutilib.misc import PauseGC
+from pyomo.core.base import (
+    SymbolMap, AlphaNumericTextLabeler, NumericLabeler,
+    Block, Constraint, Expression, Objective, Var, Set, RangeSet, Param,
+    value, minimize, Suffix)
+from pyomo.core.base.component import ComponentData
+import os, sys, subprocess, pipes, math, logging
+
+import pyomo.util.plugin
+from pyomo.opt.base import IOptSolver
+import pyutilib.services
+
+from pyutilib.misc import Options
+
+
+logger = logging.getLogger('pyomo.solvers')
+
+pyutilib.services.register_executable(name="gams")
+ 
+class GAMSSolver(pyomo.util.plugin.Plugin):
+    """A generic interface to GAMS solvers"""
+    pyomo.util.plugin.implements(IOptSolver)
+    pyomo.util.plugin.alias('gams', doc='The GAMS modeling language')
+
+    options = Options()
+
+    def available(self, exception_flag=True):
+        return pyutilib.services.registered_executable("gams") is not None
+ 
+    def solve(self, model, tee=False):
+        """
+        Solver for pyomo models using GAMS. Converts pyomo model
+        to GAMS input file, solves in GAMS, then returns solution
+        values back into pyomo model.
+
+        Uses GAMS Python API. For installation help visit:
+        https://www.gams.com/latest/docs/apis/examples_python/index.html
+
+        self.options:
+            output_filename=None:
+                GAMS model file. If None, will use model name. Kept if given.
+            symbolic_solver_labels=False:
+                Use full Pyomo component names rather than
+                shortened symbols (slower, but useful for debugging).
+            labeler=None:
+                Custom labeler option. Incompatible with symbolic_solver_labels.
+            solve=True:
+                If False, GAMSSolver will only create and keep GAMS model file.
+            solver=None:
+                If None, GAMS will use default solver for model type.
+            mtype=None:
+                Model type. If None, will chose from lp, nlp, mip, and minlp.
+            load_model=True:
+                Load results back into pyomo model.
+            print_result=False:
+                Print summary of solution to stdout.
+            keep_files=False:
+                Keep temporary files in current directory.
+                Equivalent of DebugLevel.KeepFiles.
+                If False, GAMS model file will be deleted as well.
+                Summary of temp files can be found in _gams_py_gjo0.pf
+            holdfixed=False:

```

Comment:
To clarify, you're suggesting an options argument where the user can include any options they would like to in the gms file? Essentially a string of any additional lines they would like?